process/sync/shardblock.go

```diff
@@ -369,18 +359,14 @@ func (boot *ShardBootstrap) getShardHeaderFromPool(headerHash []byte) (data.Head
 	return process.GetShardHeaderFromPool(headerHash, boot.headers)
 }
 
-func (boot *ShardBootstrap) requestMiniBlocksFromHeaderWithNonceIfMissing(shardId uint32, nonce uint64) {
+func (boot *ShardBootstrap) requestMiniBlocksFromHeaderWithNonceIfMissing(headerHash []byte, nonce uint64) {
```

Comment:
Now it is more clear that you can send headerHandler instead hash and we can remove line 369, no need to access pool again.

Reply:
done.